rustsat/src/instances/fio.rs

```diff
@@ -66,3 +75,167 @@ pub(crate) fn open_compressed_uncompressed_write<P: AsRef<Path>>(
     }
     Ok(Box::new(io::BufWriter::new(raw_writer)))
 }
+
+#[derive(Debug, PartialEq, Eq)]
+pub enum SolverOutput {
+    Sat(types::Assignment),
+    Unsat,
+    Unknown,
+}
+
+#[derive(Error, Debug)]
+pub enum SatSolverOutputError {
+    #[error("The output of the SAT solver is incorrect.")]
+    Nonsolution,
+    #[error("No solution line found in the output.")]
+    NoSline,
+    #[error("No value line found in the output.")]
+    NoVline,
+    #[error("Invalid solution line found in the output.")]
+    InvalidSLine,
+}
+
+//Parse a SAT solver's output
+pub fn parse_sat_solver_output<R: BufRead>(reader: R) -> anyhow::Result<SolverOutput> {
+    let mut is_sat = false;
+    let mut have_vline = false;
+    let mut solution = None;
+
+    for line in reader.lines() {
+        let line = &line?;
+
+        //Solution line
+        if line.starts_with('s') {
+            match line {
+                line if line.contains("UNSATISFIABLE") => return Ok(SolverOutput::Unsat),
+                line if line.contains("UNKNOWN") || line.contains("INDETERMINATE") => {
+                    return Ok(SolverOutput::Unknown)
+                }
+                line if line.contains("SATISFIABLE") => {
+                    /*if let Some(solution) = solution {
+                        return Ok(SolverOutput::Sat(solution));
+                    }*/
+                    is_sat = true;
+                }
+                _ => return Err(anyhow::anyhow!(SatSolverOutputError::InvalidSLine)),
+            }
```

Comment:
```suggestion
            // skip s and trailing whitespace
            let line = &line[1..].trim_start();
            match line {
                line if line.starts_with("UNSATISFIABLE") => return Ok(SolverOutput::Unsat),
                line if line.starts_with("UNKNOWN") || line.starts_with("INDETERMINATE") => {
                    return Ok(SolverOutput::Unknown)
                }
                line if line.starts_with("SATISFIABLE") => {
                    is_sat = true;
                }
                _ => anyhow::bail!(SatSolverOutputError::InvalidSLine),
            }
```
I think slightly more thorough checks here are reasonable. Note how your version would accept `s ERROR UNSATISFIABLE` as a valid Unsat marker.